### PR TITLE
Refactor and fix

### LINKED ISSUE #112: Disk space should only report space used for recordings
mount paths now added to OpenWebIF deviceinfo so can now check and report correct space.

### PR DESCRIPTION
**Pull request by phunkyfish**
v3.12.5
- Fixed: Large refactor for code organisation
- Fixed: Disk space, only for mounts configured for recordings - Requires OpenWebIf 1.3.5, fixes #112 

Hopefully the last large refactor ;)

### CODE REVIEW

File: src/enigma2/Admin.cpp [hunk 1]
```diff
@@ -190,9 +190,25 @@ PVR_ERROR Admin::GetDriveSpace(long long *iTotal, long long *iUsed, std::vector<
...
+
+      for (std::string& location : locations)
+      {
+        if (location.find(mount) != std::string::npos) 
+          validRecordingMount = true;
+      }
+
+      if (!validRecordingMount)
+        continue;
+    }
```
Comment (ksooo) [inline src/enigma2/Admin.cpp]: good use case for `std::find_if`.

```c++
if (!mount.empty())
{
  auto it = std::find_if(locations.begin(), locations.end(), [&mount](std::string& location) { return location.find(mount) != std::string::npos; });

  if (it == locations.end())
     continue; // no valid mount point
}
```

Even eliminates the need for the bool var.

If you want to stick with your version, please `break;`after `validRecordingMount = true;` No need to iterate further.

File: pvr.vuplus/addon.xml.in [hunk 2]
```diff
@@ -175,6 +175,10 @@
     <disclaimer lang="zh_TW">這是測試版軟體！其原創作者並無法對於以下情況負責，包含：錄影失敗，不正確的定時設定，多餘時數，或任何產生的其它不良影響...</disclaimer>
     <platform>@PLATFORM@</platform>
     <news>
+v3.12.5
+- Fixed: Large refactor for code organisation
```
Comment (ksooo) [inline pvr.vuplus/addon.xml.in]: Is refactoring a fix? I hope it does not change functionality. ;-)

Comment (phunkyfish) [reply pvr.vuplus/addon.xml.in]: I’ve just been using fixed and added. I guess neither are appropriate.

File: src/enigma2/Admin.cpp (hunk 1 above)
Comment (phunkyfish) [reply src/enigma2/Admin.cpp]: I like this, but I also like to understand my code ;)

I can't find a good reference to understand the predicate piece: 
`[&mount](std::string& location) { return location.find(mount) != std::string::npos; }`

It's mainly the `[&mount](std::string& location) ` which is not intuitive to me.

File: pvr.vuplus/addon.xml.in (hunk 2 above)
Comment (phunkyfish) [reply pvr.vuplus/addon.xml.in]: I could argue that good refactoring could be deemed as a fix ;) ...

File: src/enigma2/Admin.cpp (hunk 1 above)
Comment (phunkyfish) [reply src/enigma2/Admin.cpp]: Ah ha, lambda's. 

[<var(s) I want to pass in>](<var(s) that changes per loop>) { <your condition> }

**Comment by phunkyfish** [comment]
Ok, ready to go.

File: src/enigma2/Admin.cpp (hunk 1 above)
Comment (ksooo) [reply src/enigma2/Admin.cpp]: Yes, lambdas. 😉

**Comment by phunkyfish** [comment]
@ksooo can you approve this? 😉

File: src/enigma2/Epg.cpp [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+{
+  m_writeHandle = XBMC->OpenFileForWrite(INITIAL_EPG_READY_FILE.c_str(), true);
+  XBMC->WriteFile(m_writeHandle, "Y", 1);
+  XBMC->CloseFile(m_writeHandle);  
+}
+
+bool Epg::IsInitialEpgCompleted()
+{
+  m_readHandle = XBMC->OpenFile(INITIAL_EPG_READY_FILE.c_str(), 0);
+  byte buf[1];
```
Comment (Rechi) [inline src/enigma2/Epg.cpp]: @phunkyfish this break windows, see https://jenkins.kodi.tv/view/Addons/job/kodi-pvr/job/pvr.vuplus/job/master/2/execution/node/191/log/

**Comment by phunkyfish** [comment]
Ah, I see the problem.

I will fix as soon as I’m home from work.

**Pull request merged**